tasks/licenses.js

```diff
@@ -10,75 +8,85 @@ export default function licenses(grunt) {
     const config = this.options();
     const done = this.async();
 
-    const result = {};
-    const options = { start: process.cwd(), json: true };
-    const checkQueueLength = 2;
+    const result = [];
+    const options = {
+      start: process.cwd(),
+      json: true
+    };
 
-    function getLicenses(dependency) {
-      if (config.overrides[dependency.name]) {
-        return config.overrides[dependency.name];
-      }
+    await fromNode(cb => npm.load({ production: true }, cb));
 
-      if (dependency && dependency.licenses) {
-        return _.flatten([dependency.licenses]);
-      }
-    }
-
-    function processPackage(dependency) {
-      const pkgInfo = Object.assign({}, dependency);
-      pkgInfo.licenses = getLicenses(dependency);
-      pkgInfo.valid = (function () {
-        if (_.intersection(pkgInfo.licenses, config.licenses).length > 0) {
-          return true;
-        }
-        return false;
-      }());
-      return pkgInfo;
-    }
-
-    const allDependencies = await fromNode(cb => {
-      npmLicense.init(options, result => {
+    const packages = await fromNode(cb => {
+      npmLicense.init(options, (result, error) => {
         cb(undefined, result);
       });
     });
 
-    // Only check production NPM dependencies, not dev
-    await fromNode(cb => npm.load({production: true}, cb));
-
     const npmList = await fromNode(cb => {
       npm.commands.list([], true, (a, b, npmList) => cb(undefined, npmList));
     });
 
-    // Recurse npm --production --json ls output, create array of dependency
-    // objects, each with a name prop formatted as 'package@version' and a
-    // path prop leading back to the root dependency.
-    function getDependencies(dependencies, path, list = []) {
-      Object.keys(dependencies).forEach(name => {
-        const dependency = dependencies[name];
-        const newPath = path ? `${path} -> ${dependency.from}` : dependency.from;
-        list.push({
-          path: newPath,
-          name: name + '@' + dependency.version
-        });
+    function dependencies(deps, path = [], list = {}) {
+      Object.keys(deps).forEach(name => {
+        let dependency = deps[name];
+        let displayName = `${name}@${dependency.version}`;
+
+        list[displayName] = {
+          name: name,
+          version: dependency.version,
+          path: path
+        };
 
         if (dependency.dependencies) {
-          getDependencies(dependency.dependencies, newPath, list);
+          dependencies(dependency.dependencies, path.concat(displayName), list);
```

Comment:
I think naming functions/methods as verbs makes code more readable because it follows the "verb noun" format of most sentences. Any reason to change this?